### PR TITLE
refactor(DIS):refactor resource_huaweicloud_dis_stream_v2 to new SDK

### LINKED ISSUE #1504: [DIS] refactor resource_huaweicloud_dis_stream_v2 to new SDK
refactor resource_huaweicloud_dis_stream_v2 to new SDK

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
refactor resource_huaweicloud_dis_stream_v2 to new SDK

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1504 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dis' TESTARGS='-run=TestAccResourceDisStream'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dis -v -run=TestAccResourceDisStream -timeout 360m -parallel 4
=== RUN   TestAccResourceDisStream_basic
=== PAUSE TestAccResourceDisStream_basic
=== RUN   TestAccResourceDisStream_all
=== PAUSE TestAccResourceDisStream_all
=== CONT  TestAccResourceDisStream_basic
=== CONT  TestAccResourceDisStream_all
--- PASS: TestAccResourceDisStream_all (42.54s)
--- PASS: TestAccResourceDisStream_basic (47.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dis       47.076s
```

### CODE REVIEW

File: docs/resources/dis_stream.md [hunk 1]
```diff
@@ -44,11 +46,11 @@ The following arguments are supported:
 * `region` - (Optional, String, ForceNew) The region in which to create the DIS stream resource. If omitted, the
   provider-level region will be used. Changing this creates a new DIS Stream resource.
 
-* `retention_period` - (Optional, Int, ForceNew) The number of hours for which data from the stream will be retained in
-  DIS. Value range: 24 to 72.Unit: hour. Changing this parameter will create a new resource.
+* `retention_period` - (Optional, Int) The number of hours for which data from the stream will be retained in DIS.
+  Value range: `24` to `72`.Unit: `hour`. Default:`24`.
```
Comment (Lance52259) [inline docs/resources/dis_stream.md]: s/`72`.Unit/`72`. Unit

File: docs/resources/dis_stream.md [hunk 2]
```diff
@@ -88,3 +83,15 @@ In addition to all arguments above, the following attributes are exported:
 * `readable_partition_count` - Total number of readable partitions (including partitions in ACTIVE state only).
 
 * `writable_partition_count` - Total number of writable partitions (including partitions in ACTIVE and DELETED states).
+
+* `status` - Status of stream: `CREATING`,`RUNNING`,`TERMINATING`,`TERMINATED`,`FROZEN`
+
+* `stream_id` - Specifies a stream ID in UUID format
```
Comment (Lance52259) [inline docs/resources/dis_stream.md]: missing period (.)

File: docs/resources/dis_stream.md [hunk 3]
```diff
@@ -88,3 +83,15 @@ In addition to all arguments above, the following attributes are exported:
...
+
+## Import
+
+Dis stream can be imported by  `stream_name`. For example,
```
Comment (Lance52259) [inline docs/resources/dis_stream.md]: only one blank space between `by` and `stream_name`.

File: huaweicloud/services/dis/resource_huaweicloud_dis_stream.go [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+const DIS_SYS_TAG_KEY_ENTERPRISE_PROJECT_ID = "_sys_enterprise_project_id"
```
Comment (Lance52259) [inline huaweicloud/services/dis/resource_huaweicloud_dis_stream.go]: The value can be declared in common as a common constant for reference and the format of constant should be in `UpperCamelCase`.

File: huaweicloud/services/dis/resource_huaweicloud_dis_stream.go [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+				Optional: true,
+				Default:  24,
+				ForceNew: true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+					if v < 24 || v > 72 {
+						errs = append(errs, fmtp.Errorf("%q must be between 24 and 72 inclusive, got: %d", key, v))
+					}
+					return
+				},
```
Comment (Lance52259) [inline huaweicloud/services/dis/resource_huaweicloud_dis_stream.go]: use validation.IntBetween(24, 72)

File: huaweicloud/utils/type_convert.go [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package utils
+
+// returns a pointer to the bool value
+func Bool(v bool) *bool {
+	return &v
+}
```
Comment (ShiChangkuo) [inline huaweicloud/utils/type_convert.go]: I think this function is rare used, and we should support more type if necessary.
So, I think you can update it in another PR.

File: huaweicloud/config/config.go [hunk 7]
```diff
@@ -832,6 +832,10 @@ func (c *Config) DisV2Client(region string) (*golangsdk.ServiceClient, error) {
 	return c.NewServiceClient("disv2", region)
 }
 
+func (c *Config) DisV3Client(region string) (*golangsdk.ServiceClient, error) {
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go:839]: the testing for this client is missing

File: huaweicloud/services/dis/resource_huaweicloud_dis_stream.go [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"auto_scale_min_partition_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
```
Comment (ShiChangkuo) [inline huaweicloud/services/dis/resource_huaweicloud_dis_stream.go]: suggest to add **RequiredWith** attribute

**Review by ShiChangkuo — CHANGES_REQUESTED** [review]

File: huaweicloud/config/config.go (hunk 7 above)
Comment (hmbig2) [reply huaweicloud/config/config.go:839]:
```
    endpoints_test.go:817: MRS v1.1 endpoint:    https://mrs.cn-north-4.myhuaweicloud.com/v1.1/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:829: MRS v2 endpoint:      https://mrs.cn-north-4.myhuaweicloud.com/v2/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:841: SMN endpoint:         https://smn.cn-north-4.myhuaweicloud.com/v2/0970dd7a1300f5672ff2c003c60ae115/notifications/
    endpoints_test.go:852: cdm endpoint:         https://cdm.cn-north-4.myhuaweicloud.com/v1.1/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:863: dis endpoint:         https://dis.cn-north-4.myhuaweicloud.com/v2/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:874: dis v3 endpoint:      https://dis.cn-north-4.myhuaweicloud.com/v3/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:885: cloudtable endpoint:  https://cloudtable.cn-north-4.myhuaweicloud.com/v2/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:896: cloudStream endpoint:         https://cs.cn-north-4.myhuaweicloud.com/v1.0/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:907: css endpoint:         https://css.cn-north-4.myhuaweicloud.com/v1.0/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:918: dli endpoint:         https://dli.cn-north-4.myhuaweicloud.com/v1.0/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:929: dws endpoint:         https://dws.cn-north-4.myhuaweicloud.com/v1.0/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:940: dws V2 endpoint:      https://dws.cn-north-4.myhuaweicloud.com/v2/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:951: ges endpoint:         https://ges.cn-north-4.myhuaweicloud.com/v1.0/0970dd7a1300f5672ff2c003c60ae115/
    endpoints_test.go:962: mls endpoint:         https://mls.cn-north-4.myhuaweicloud.com/v1.0/0970dd7a1300f5672ff2c003c60ae115/
--- PASS: TestAccServiceEndpoints_EnterpriseIntelligence (1.34s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       1.457s
```

File: docs/resources/dis_stream.md [hunk 9]
```diff
@@ -44,11 +46,11 @@ The following arguments are supported:
 * `region` - (Optional, String, ForceNew) The region in which to create the DIS stream resource. If omitted, the
   provider-level region will be used. Changing this creates a new DIS Stream resource.
 
-* `retention_period` - (Optional, Int, ForceNew) The number of hours for which data from the stream will be retained in
-  DIS. Value range: 24 to 72.Unit: hour. Changing this parameter will create a new resource.
+* `retention_period` - (Optional, Int) The number of hours for which data from the stream will be retained in DIS.
```
Comment (ShiChangkuo) [inline docs/resources/dis_stream.md]: this parameter is **ForceNew** in schema

**Pull request merged**